plugins/azure/fix_plugin_azure/azure_client.py

```diff
@@ -21,6 +23,28 @@
 from fix_plugin_azure.config import AzureCredentials
 from fixlib.types import Json
 
+log = logging.getLogger("fix.plugins.azure")
+
+
+class MetricRequestError(HttpResponseError):
+    """Error raised when there's an issue retrieving metric data from the Azure API."""
+
+
+def is_retryable_exception(e: Exception) -> bool:
+    # If we receive a metric request error, then repeat the request
+    if isinstance(e, MetricRequestError):
+        log.debug(f"Azure Metric request error occured, retrying: {e}")
+        return True
```

Comment:
An exception of this class is created in the exception handler and then catched here again.
Maybe it is simpler to match the original exception here?

Reply:
I use the `MetricRequestError` class to match the metric error exactly and retry the request for it. If we compare the original exception (`HttpResponseError`) with error_code as `BadRequest`, then for other resources (except metrics resources) having the same error code, we will try to get them again. And it doesn't suit us